### PR TITLE
issues #10966, Add test class for shardingsphere-shadow project

### LINKED ISSUE #10966: Add unit test for shardingsphere-shadow
Hi community,

This issue is for #10887.

## Aim
Add unit test for shardingsphere-shadow project to test its public functions.

## Basic Qualifications
- Java  
- Maven  
- Junit.Test

## Classes

Add unit test for classes to test its public functions.

- [ ] ShadowSQLRewriteContextDecorator
- [ ] ShadowInsertValueParameterRewriter
- [ ] ShadowPredicateParameterRewriter
- [ ] ShadowParameterRewriterBuilder
- [ ] RemoveShadowColumnTokenGenerator
- [ ] ShadowInsertValuesTokenGenerator
- [ ] ShadowPredicateColumnTokenGenerator
- [ ] ShadowInsertValuesToken
- [ ] ShadowTokenGenerateBuilder

## Examples FYI
- ShadowRuleBuilderTest

### PR DESCRIPTION
**Pull request by soulasuna**
Fixes #10966.

Add test class for shardingsphere-shadow project.

### CODE REVIEW

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/ShadowPredicateColumnTokenGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Test(expected = ClassCastException.class)
+    public void generateSQLTokensTest(){
+        shadowPredicateColumnTokenGenerator.generateSQLTokens(sqlStatementContext);
+    }
+
+
+
+
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/ShadowPredicateColumnTokenGeneratorTest.java:39]: Please remove all the redundant blank lines.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/ShadowPredicateColumnTokenGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * test for ShadowPredicateColumnTokenGenerator
+ */
+public class ShadowPredicateColumnTokenGeneratorTest {
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/ShadowPredicateColumnTokenGeneratorTest.java:15]: All the test classes are expected to be defined as the final classes.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/ShadowPredicateColumnTokenGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.apache.shardingsphere.shadow.rewrite.token.generator.impl.ShadowPredicateColumnTokenGenerator;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.SQLStatement;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * test for ShadowPredicateColumnTokenGenerator
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/ShadowPredicateColumnTokenGeneratorTest.java:13]: Please capitalize the first letter and check other classes in the same place.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowInsertValuesTokenGenerator.java [hunk 4]
```diff
@@ -102,15 +102,15 @@ private void removeValueToken(final InsertValue insertValueToken, final InsertSt
...
+        int result = Integer.MAX_VALUE;
         for (InsertValuesSegment each : segments) {
             result = Math.min(result, each.getStartIndex());
         }
         return result;
     }
     
     private int getStopIndex(final Collection<InsertValuesSegment> segments) {
-        int result = segments.iterator().next().getStopIndex();
+        int result = Integer.MIN_VALUE;
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowInsertValuesTokenGenerator.java:113]: Why is it changed to `MIN_VALUE`?

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowInsertValueParameterRewriter.java [hunk 5]
```diff
@@ -33,7 +33,7 @@
 public final class ShadowInsertValueParameterRewriter extends ShadowParameterRewriter<InsertStatementContext> {
     
     @Override
-    protected boolean isNeedRewriteForShadow(final SQLStatementContext sqlStatementContext) {
+    public boolean isNeedRewriteForShadow(final SQLStatementContext sqlStatementContext) {
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowInsertValueParameterRewriter.java:36]: Why do you use `public`?

**Review by tristaZero — CHANGES_REQUESTED** [review]

**Comment by soulasuna** [comment]
@tristaZero Thank you for your advice. I fixed it

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowPredicateParameterRewriter.java [hunk 6]
```diff
@@ -36,9 +36,10 @@
     protected boolean isNeedRewriteForShadow(final SQLStatementContext sqlStatementContext) {
         return true;
     }
-    
+
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowPredicateParameterRewriter.java:39]: Please, keep the indent as it.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowPredicateParameterRewriter.java [hunk 7]
```diff
@@ -36,9 +36,10 @@
...
     @Override
     public void rewrite(final ParameterBuilder parameterBuilder, final SQLStatementContext sqlStatementContext, final List<Object> parameters) {
+
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowPredicateParameterRewriter.java:42]: Please remove the blank line.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/condition/ShadowConditionTest.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public final class ShadowConditionTest {
+
+    @Test
+    public void getValuesTest() {
+        ShadowCondition shadowCondition = new ShadowCondition("a", 0, 10,
+                new LiteralExpressionSegment(0, 10, "result"));
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/condition/ShadowConditionTest.java:32]: Please join the lines.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/context/ShadowSQLRewriteContextDecoratorTest.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+    private ShadowSQLRewriteContextDecorator shadowSQLRewriteContextDecorator;
+
+    @Before
+    public void init() {
+        shadowSQLRewriteContextDecorator = new ShadowSQLRewriteContextDecorator();
+    }
+
+    @Test
+    public void getOrderTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/context/ShadowSQLRewriteContextDecoratorTest.java:37]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/context/ShadowSQLRewriteContextDecoratorTest.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        assertThat(shadowSQLRewriteContextDecorator.getOrder(), is(60));
+    }
+
+    @Test
+    public void getTypeClassTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/context/ShadowSQLRewriteContextDecoratorTest.java:42]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/parameter/ShadowParameterRewriterBuilderTest.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+    private ShadowParameterRewriterBuilder shadowParameterRewriterBuilder;
+
+    @Before
+    public void init() {
+        shadowParameterRewriterBuilder = new ShadowParameterRewriterBuilder(mock(ShadowRule.class));
+    }
+    
+    @Test
+    public void getParameterRewritersTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/parameter/ShadowParameterRewriterBuilderTest.java:39]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowInsertValueParameterRewriterTest.java [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    private ShadowRule mockShadowRule(final String shadowColumn) {
+        ShadowRule shadowRule = mock(ShadowRule.class);
+        when(shadowRule.getColumn()).thenReturn(shadowColumn);
+        return shadowRule;
+    }
+
+    @Test
+    public void isNeedRewriteForShadowTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowInsertValueParameterRewriterTest.java:63]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowInsertValueParameterRewriterTest.java [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+        assertTrue(shadowInsertValueParameterRewriter.isNeedRewriteForShadow(insertStatementContext));
+    }
+
+    @Test
+    public void rewriteTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowInsertValueParameterRewriterTest.java:68]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowPredicateParameterRewriterTest.java [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+
+    private ShadowPredicateParameterRewriter shadowPredicateParameterRewriter;
+
+    @Before
+    public void init() {
+        shadowPredicateParameterRewriter = new ShadowPredicateParameterRewriter();
+    }
+
+    @Test
+    public void isNeedRewriteForShadowTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowPredicateParameterRewriterTest.java:40]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowPredicateParameterRewriterTest.java [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+        assertTrue(shadowPredicateParameterRewriter.isNeedRewriteForShadow(mock(SQLStatementContext.class)));
+    }
+
+    @Test
+    public void rewriteTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/parameter/impl/ShadowPredicateParameterRewriterTest.java:45]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/ShadowTokenGenerateBuilderTest.java [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+
+    private ShadowTokenGenerateBuilder shadowTokenGenerateBuilder;
+
+    @Before
+    public void init() {
+        shadowTokenGenerateBuilder = new ShadowTokenGenerateBuilder(mock(ShadowRule.class));
+    }
+
+    @Test
+    public void getSQLTokenGeneratorsTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/ShadowTokenGenerateBuilderTest.java:38]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/RemoveShadowColumnTokenGeneratorTest.java [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    private ShadowRule mockShadowRule(final String shadowColumn) {
+        ShadowRule shadowRule = mock(ShadowRule.class);
+        when(shadowRule.getColumn()).thenReturn(shadowColumn);
+        return shadowRule;
+    }
+
+    @Test
+    public void isGenerateSQLTokenForShadowTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/RemoveShadowColumnTokenGeneratorTest.java:86]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/RemoveShadowColumnTokenGeneratorTest.java [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+        assertTrue(removeShadowColumnTokenGenerator.isGenerateSQLTokenForShadow(insertStatementContext));
+    }
+
+    @Test
+    public void generateSQLTokensTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/RemoveShadowColumnTokenGeneratorTest.java:91]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowInsertValuesTokenGeneratorTest.java [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    private ShadowRule mockShadowRule(final String shadowColumn) {
+        ShadowRule shadowRule = mock(ShadowRule.class);
+        when(shadowRule.getColumn()).thenReturn(shadowColumn);
+        return shadowRule;
+    }
+
+    @Test
+    public void isGenerateSQLTokenForShadowTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowInsertValuesTokenGeneratorTest.java:82]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowInsertValuesTokenGeneratorTest.java [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+        Assert.assertTrue(shadowInsertValuesTokenGenerator.isGenerateSQLTokenForShadow(insertStatementContext));
+    }
+
+    @Test
+    public void generateSQLTokenTest() {
+        Assert.assertNotNull(shadowInsertValuesTokenGenerator.generateSQLToken(insertStatementContext));
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowInsertValuesTokenGeneratorTest.java:88]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowPredicateColumnTokenGeneratorTest.java [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+    private SQLStatementContext<? extends SQLStatement> sqlStatementContext;
+
+    @Before
+    public void init() {
+        shadowPredicateColumnTokenGenerator = new ShadowPredicateColumnTokenGenerator();
+        sqlStatementContext = mock(SQLStatementContext.class);
+    }
+
+    @Test
+    public void isGenerateSQLTokenForShadow() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowPredicateColumnTokenGeneratorTest.java:41]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowPredicateColumnTokenGeneratorTest.java [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+        Assert.assertFalse(shadowPredicateColumnTokenGenerator.isGenerateSQLTokenForShadow(sqlStatementContext));
+    }
+
+    @Test(expected = ClassCastException.class)
+    public void generateSQLTokensTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowPredicateColumnTokenGeneratorTest.java:46]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/pojo/ShadowInsertValuesTokenTest.java [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+
+    private ShadowInsertValuesToken shadowInsertValuesToken;
+
+    @Before
+    public void init() {
+        shadowInsertValuesToken = new ShadowInsertValuesToken(0, 10);
+    }
+
+    @Test
+    public void toStringTest() {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/pojo/ShadowInsertValuesTokenTest.java:36]: Name the test methods like `assertXXX`.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/ShadowTokenGenerateBuilderTest.java [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+
+import org.apache.shardingsphere.shadow.rule.ShadowRule;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class ShadowTokenGenerateBuilderTest {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/ShadowTokenGenerateBuilderTest.java:28]: Add `final` modifier.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowInsertValuesTokenGeneratorTest.java [hunk 25]
```diff
@@ -0,0 +1,91 @@
...
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ShadowInsertValuesTokenGeneratorTest {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowInsertValuesTokenGeneratorTest.java:36]: Add `final` modifier.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowPredicateColumnTokenGeneratorTest.java [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+
+import org.apache.shardingsphere.infra.binder.statement.SQLStatementContext;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.SQLStatement;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.mock;
+
+public class ShadowPredicateColumnTokenGeneratorTest {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/generator/impl/ShadowPredicateColumnTokenGeneratorTest.java:28]: Add `final` modifier.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/pojo/ShadowInsertValuesTokenTest.java [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+
+package org.apache.shardingsphere.shadow.rewrite.token.pojo;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class ShadowInsertValuesTokenTest {
```
Comment (TeslaCN) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/rewrite/token/pojo/ShadowInsertValuesTokenTest.java:26]: Add `final` modifier.

**Review by tristaZero — APPROVED** [review]
Nice work on your first PR. @soulasuna

**Comment by soulasuna** [comment]
Thanks to  @tristaZero @TeslaCN  for their help

**Pull request merged**